network/ocsp.go

```diff
@@ -0,0 +1,334 @@
+		ocspFromCertUseStr:    ocspFromCertUse,
+		ocspFromCertTrustStr:  ocspFromCertTrust,
+		ocspFromCertPreferStr: ocspFromCertPrefer,
+		ocspFromCertIgnoreStr: ocspFromCertIgnore,
+	}
+
+	fromCertVal, ok := fromCertValValues[fromCert]
+	if !ok {
+		return nil, ErrInvalidConfigOCSPFromCert
+	}
+
+	if requiredVal == ocspRequiredAll && uri == "" {
+		return nil, ErrInvalidConfigAllRequiresURI
+	}
+
+	if uri != "" {
+		log.Debugf("OCSP: Using server '%s'", uri)
+
+		httpClient := &http.Client{}
+		_, err := httpClient.Head(uri)
+		if err != nil {
+			log.WithError(err).WithField("uri", uri).Warnln("OCSP: Cannot reach configured server")
+			// TODO return error after issues with failing tests are fixed;
+			//      the issue is the same as in NewCRLConfig()
+			// return nil, errors.New("Cannot reach configured OCSP server")
+		}
+	}
+
+	switch requiredVal {
+	case ocspRequiredYes:
+		log.Debugln("OCSP: At least one OCSP server should confirm certificate validity")
+	case ocspRequiredNo:
+		log.Debugln("OCSP: Allowing certificates not known by OCSP server")
+	case ocspRequiredAll:
+		log.Debugln("OCSP: Requiring positive response from all OCSP servers")
+	}
+
+	switch fromCertVal {
+	case ocspFromCertUse:
+		log.Debugln("OCSP: using servers described in certificates if nothing passed via command line")
+	case ocspFromCertTrust:
+		log.Debugln("OCSP: trusting responses from OCSP servers listed in certificates")
+	case ocspFromCertPrefer:
+		log.Debugln("OCSP: server from certificate will be prioritized over one from command line")
+	case ocspFromCertIgnore:
+		log.Debugln("OCSP: ignoring OCSP servers described in certificates")
+	}
+
+	return &OCSPConfig{url: uri, required: requiredVal, fromCert: fromCertVal}, nil
+}
+
+// OCSPClient is used to perform OCSP queries to some URI
+type OCSPClient interface {
+	// Query generates OCSP request about specified certificate, sends it to server and returns the response
+	Query(commonName string, clientCert, issuerCert *x509.Certificate, ocspServerURL string) (*ocsp.Response, error)
+}
+
+// DefaultOCSPClient is a default implementation of OCSPClient
+type DefaultOCSPClient struct{}
+
+// Query generates OCSP request about specified certificate, sends it to server and returns the response
+func (c DefaultOCSPClient) Query(commonName string, clientCert, issuerCert *x509.Certificate, ocspServerURL string) (*ocsp.Response, error) {
+	opts := &ocsp.RequestOptions{Hash: crypto.SHA256}
+	buffer, err := ocsp.CreateRequest(clientCert, issuerCert, opts)
+	if err != nil {
+		return nil, err
+	}
+	httpRequest, err := http.NewRequest(http.MethodPost, ocspServerURL, bytes.NewBuffer(buffer))
+	if err != nil {
+		return nil, err
+	}
+	ocspURL, err := url.Parse(ocspServerURL)
+	if err != nil {
+		return nil, err
+	}
+	httpRequest.Header.Add("Content-Type", "application/ocsp-request")
+	httpRequest.Header.Add("Accept", "application/ocsp-response")
+	httpRequest.Header.Add("host", ocspURL.Host)
+	httpClient := &http.Client{}
```

Comment:
why not initialize it once as a field of DefaultOCSPClient with configured timeouts?

Reply:
Done, now `http.Client` is created and stored in OCSPClient, with configured timeout of 15s